lnode/syncrandaddr.go

```diff
@@ -0,0 +1,299 @@
+package lnode
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/nknorg/nkn/v2/chain/pool"
+	"github.com/nknorg/nkn/v2/common"
+	"github.com/nknorg/nkn/v2/config"
+	"github.com/nknorg/nkn/v2/node"
+	"github.com/nknorg/nkn/v2/pb"
+	"github.com/nknorg/nkn/v2/transaction"
+	"github.com/nknorg/nkn/v2/util"
+	"github.com/nknorg/nkn/v2/util/log"
+)
+
+// Number of random neighbors to sync rand addr
+const NumNbrToSyncRandAddr = 1
+
+func (localNode *LocalNode) StartSyncRandAddrTxn() {
+	for {
+		dur := util.RandDuration(config.ConsensusDuration/2, 0.4)
+		time.Sleep(dur)
+		localNode.SyncRandAddrTxn()
+	}
+}
+
+// sync rand addr process
+func (localNode *LocalNode) SyncRandAddrTxn() {
+
+	log.Info("Begin sync rand addr now.........")
+
+	neighbors := localNode.GetNeighbors(nil)
+	neighborCount := len(neighbors)
+	if neighborCount <= 0 { // don't have neighbor yet
+		return
+	}
+
+	if NumNbrToSyncRandAddr < neighborCount {
+		// choose rand neighbors
+		r := rand.New(rand.NewSource(time.Now().UnixNano()))
+		r.Shuffle(len(neighbors), func(i, j int) { neighbors[i], neighbors[j] = neighbors[j], neighbors[i] })
+
+		neighbors = neighbors[:NumNbrToSyncRandAddr]
+	}
+
+	for _, neighbor := range neighbors {
+
+		go func(nbr *node.RemoteNode) {
+
+			addrNonceReply, err := localNode.requestAddrNonce(nbr)
+			if err != nil {
+				log.Warningf("requestAddrNonce to neighbor %v error: %v\n", nbr.GetAddr(), err)
+				return
+			}
+
+			var addr160 common.Uint160
+			addr160.SetBytes(addrNonceReply.Address)
+
+			// here we get next expected nonce = last nonce + 1
+			myNonce, err := localNode.TxnPool.GetNonceByTxnPool(addr160)
+			if err != nil {
+				log.Info("localNode.TxnPool.GetNonceByTxnPool err: ", err)
```

Comment:
This is normal (local txpool doesn't have the txn), no need to spam it in info log.
Also, in this case we should use ledger nonce `chain.DefaultLedger.Store.GetNonce()`

Reply:
yes. fixed.